src/PlugWallet/ledger.ts

```diff
@@ -0,0 +1,177 @@
+    };
+
+    public async burnXTC(args: { to: string; amount: string }) {
+      throw new Error('Burn XTC not supported on Ledger');
+      return this.executeWithDelegation(super.burnXTC, args);
+    };  
+    
+    public async send(args: {
+      to: string;
+      amount: string;
+      canisterId: string;
+      opts?: TokenInterfaces.SendOpts;
+    }): Promise<TokenInterfaces.SendResponse> {
+      if (args.canisterId !== LEDGER_CANISTER_ID) {
+        throw new Error(`Token ${args.canisterId} not supported on Ledger`)
+      }
+      const to = validateAccountId(args.to) ? args.to : getAccountId(Principal.fromText(args.to));
+
+      if(!validateAccountId(to)) {
+        throw new Error('Invalid Account ID')
+      }
+
+      const canisterId = Principal.fromText(LEDGER_CANISTER_ID);
+
+      const agent = createAgent({defaultIdentity: this.ledgerIdentity});
+
+      const amount = args.amount; // 0.1 ICP
+      const memo = args.opts?.memo || BigInt(0);
+      const maxFee = args.opts?.fee || BigInt(10000);
+
+      const toPb = AccountIdentifier.fromObject({ hash: Uint8Array.from(Buffer.from(to, 'hex')) });
+      const amountPb = ICPTs.fromObject({ e8s: amount });
+      const memoPb = Memo.fromObject({ memo: memo.toString(10) });
+      const payment = Payment.fromObject({ receiverGets: amountPb});
+      const maxFeePb = ICPTs.fromObject({ e8s: maxFee.toString(10) });
+
+      const request = SendRequest.fromObject({to: toPb, payment: payment, memo: memoPb, maxFee: maxFeePb});
+    
+      const submitResponse = await agent.call(canisterId, {
+        methodName: 'send_pb',
+        arg: blobFromUint8Array(SendRequest.encode(request).finish()),
+        effectiveCanisterId: canisterId,
+      });
+    
+      if (!submitResponse.response.ok) {
+        throw new Error(
+          [
+            "Call failed:",
+            `  Method: send_pb`,
+            `  Canister ID: ${canisterId}`,
+            `  Request ID: ${submitResponse.requestId}`,
+            `  HTTP status code: ${submitResponse.response.status}`,
+            `  HTTP status text: ${submitResponse.response.statusText}`,
+          ].join("\n")
+        );
+      }
+    
+      const blob = await polling.pollForResponse(
+        agent,
+        canisterId,
+        submitResponse.requestId,
+        polling.defaultStrategy()
+      );
+
+      const { height } =  BlockHeight.decode(blob).toJSON();
+    
+      return {
+        height
+      }
+    };
+
+    public async setReverseResolvedName (args: {
+      name: string;
+    }): Promise<string> {
+      throw new Error('Set reverse resolve name not supported on Ledger');
+      return this.executeWithDelegation(super.setReverseResolvedName, args);
+    };
+
+    public async addContact(args: { contact: Address; }): Promise<boolean> {
+      throw new Error('Add contact not supported on Ledger');
+      return this.executeWithDelegation(super.addContact, args);
```

Comment:
Im not sure why are both throw and return here, the return line never reachs out, right? 🤔